go/vt/vtgate/evalengine/compiler_test.go

```diff
@@ -119,7 +119,7 @@ func TestCompilerReference(t *testing.T) {
 			var supported, total int
 			env := evalengine.EmptyExpressionEnv(venv)
 
-			tc.Run(func(query string, row []sqltypes.Value) {
+			tc.Run(func(query string, row []sqltypes.Value, skipCollationCheck bool) {
```

Comment:
shouldn't we use `skipCollationCheck` here?

Reply:
@systay This currently doesn't do collations checks anyway, so I don't think it really matters.

Reply:
@systay This did remind me that we actually should validate the collation. We still ignore this flag though, since our evalengine and compiler should always match.

The ignore collation flag is for where explicitly don't 100% match MySQL (due to bugs, arguably broken behavior etc).